File: companies/AssetMind/codebase/assetmind-enterprise/src/services/audit_service.py
"""
Audit Service
Activity logs
Port: 5223
"""

from fastapi import FastAPI
from typing import Dict, List, Any
from datetime import datetime
import logging

logger = logging.getLogger(__name__)

app = FastAPI(title="Audit Service", version="1.0.0")


class AuditService:
    """Audit logging service"""

    def __init__(self, config: Dict[str, Any] = None):
        self.config = config or {}
        self.name = "Audit Service"
        self.port = 5223
        self.logs = []

    async def log_activity(
        self,
        user_id: str,
        action: str,
        resource: str
    ) -> Dict[str, Any]:
        """Log an activity"""
        log_entry = {
            "log_id": f"log_{datetime.utcnow().timestamp()}",
            "user_id": user_id,
            "action": action,
            "resource": resource,
            "timestamp": datetime.utcnow().isoformat(),
        }
        self.logs.append(log_entry)
        return log_entry


service = AuditService()


@app.get("/health")
async def health():
    return {"status": "healthy", "service": "Audit", "port": 5223}


@app.post("/api/v1/log")
async def log_activity(request: Dict[str, Any]):
    return await service.log_activity(
        request["user_id"],
        request["action"],
        request["resource"]
    )

File: companies/KHAIRMOVE/rez-ride/src/services/audit.service.ts
import { Injectable, Logger } from '@nestjs/common';

/**
 * Audit Service - Compliance logging
 */
@Injectable()
export class AuditService {
  private logger = new Logger('AuditService');
  private logs: AuditLog[] = [];

  async log(action: string, userId: string, data: any): Promise<void> {
    const entry = { action, userId, data, timestamp: new Date(), ip: '0.0.0.0' };
    this.logs.push(entry);
    this.logger.log(`${action}: ${userId}`);
  }

  async query(filters: Partial<AuditLog>): Promise<AuditLog[]> {
    return this.logs.filter(l =>
      (!filters.action || l.action === filters.action) &&
      (!filters.userId || l.userId === filters.userId)
    );
  }
}

export interface AuditLog {
  action: string;
  userId: string;
  data: any;
  timestamp: Date;
  ip: string;
}
